src/test/java/uk/gov/companieshouse/documentsigningrequestconsumer/EnvironmentVariablesCheckerTest.java

```diff
@@ -36,119 +35,122 @@
+        populateAllVariablesExceptOneAndAssertSomethingMissing(DOCUMENT_SIGNING_REQUEST_CONSUMER_PORT.getName());
     }
 
     @DisplayName("returns false if TOPIC is missing")
     @Test
     void checkEnvironmentVariablesAllPresentReturnsFalseIfTopicMissing() {
-        populateAllVariablesExceptOneAndAssertSomethingMissing(TOPIC);
+        populateAllVariablesExceptOneAndAssertSomethingMissing(TOPIC.getName());
     }
 
     @DisplayName("returns false if API_URL is missing")
     @Test
     void checkEnvironmentVariablesAllPresentReturnsFalseIfApiUrlMissing() {
-        populateAllVariablesExceptOneAndAssertSomethingMissing(API_URL);
+        populateAllVariablesExceptOneAndAssertSomethingMissing(API_URL.getName());
     }
 
     @DisplayName("returns false if PAYMENTS_API_URL is missing")
     @Test
     void checkEnvironmentVariablesAllPresentReturnsFalseIfPaymentsApiUrlMissing() {
-        populateAllVariablesExceptOneAndAssertSomethingMissing(PAYMENTS_API_URL);
+        populateAllVariablesExceptOneAndAssertSomethingMissing(PAYMENTS_API_URL.getName());
     }
 
     @DisplayName("returns false if CHS_API_KEY is missing")
     @Test
     void checkEnvironmentVariablesAllPresentReturnsFalseIfChsApiKeyMissing() {
-        populateAllVariablesExceptOneAndAssertSomethingMissing(CHS_API_KEY);
+        populateAllVariablesExceptOneAndAssertSomethingMissing(CHS_API_KEY.getName());
     }
 
     @DisplayName("returns false if INTERNAL_API_URL is missing")
     @Test
     void checkEnvironmentVariablesAllPresentReturnsFalseIfInternalApiUrlMissing() {
-        populateAllVariablesExceptOneAndAssertSomethingMissing(INTERNAL_API_URL);
+        populateAllVariablesExceptOneAndAssertSomethingMissing(INTERNAL_API_URL.getName());
     }
 
     @DisplayName("returns false if PREFIX is missing")
     @Test
     void checkEnvironmentVariablesAllPresentReturnsFalseIfPrefixMissing() {
-        populateAllVariablesExceptOneAndAssertSomethingMissing(PREFIX);
-    }
-    private void populateAllVariablesExceptOneAndAssertSomethingMissing(
-            final EnvironmentVariablesChecker.RequiredEnvironmentVariables excludedVariable) {
-        stream(EnvironmentVariablesChecker.RequiredEnvironmentVariables.values()).forEach(variable -> {
-            if (variable != excludedVariable) {
-                environmentVariables.set(variable.getName(), TOKEN_VALUE);
-            }
+        populateAllVariablesExceptOneAndAssertSomethingMissing(PREFIX.getName());
+    }
+    private void populateAllVariablesExceptOneAndAssertSomethingMissing(String token){
+        // Loop through each environment variables
+        // Remove the current environment variable
+        // Assert that not all required environment variables are present
+        // Repopulate the environment variable for the next iteration
+        stream(EnvironmentVariablesChecker.RequiredEnvironmentVariables.values()).forEach(envVariable -> {
```

Comment:
Why is this in a loop? You can call each of .remote(), the assertion, and the .set() directly

Reply:
this is more succinct and without change to the functionality whatsoever

Reply:
I misread the last line where it's setting `envVariable` back to a value
I see now this test is ensuring `allRequiredEnvironmentVariablesPresent` correctly triggers. It's doing it in a strange way though, removing `token` then setting `envVariable` on every loop. Would make more sense to remove `token` and instead do
```
            environmentVariables.remove(envVariable);
            assertFalse(environmentVariablesChecker.allRequiredEnvironmentVariablesPresent());
            environmentVariables.set(envVariable.getName(), TOKEN_VALUE);
```
But a minor complaint